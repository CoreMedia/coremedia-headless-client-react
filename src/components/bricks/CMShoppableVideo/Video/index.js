// @flow
import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { Teaser } from '../../CMVideo';

export const VideoBox = styled.div`
  opacity: ${props => (props.show ? 1 : 0)};
  transition: opacity 0.6s linear 0s;
`;

VideoBox.displayName = 'VideoBox';

type Props = {
  show: boolean,
  link: string,
  playing?: boolean,
  loop?: boolean,
  muted?: boolean,
  controls?: boolean,
  pictureLink?: string,
  pictureTitle?: string,
  pictureAlt?: string,
  handleProgress?: (playedSeconds: number, played: number) => void,
  handlePlay?: () => void,
  handlePause?: () => void,
  handleEnded?: () => void,
  handleDuration?: (duration: number) => void,
};

const Video = ({
  show,
  link,
  playing,
  loop,
  muted,
  controls,
  pictureLink,
  pictureTitle,
  pictureAlt,
  handleProgress,
  handlePlay,
  handlePause,
  handleEnded,
  handleDuration,
}: Props) => (
  <VideoBox show={show}>
    <Teaser
      link={link}
      playing={playing}
      loop={loop}
      muted={muted}
      controls={controls}
      pictureLink={pictureLink}
      pictureTitle={pictureTitle}
      pictureAlt={pictureAlt}
      handleProgress={handleProgress}
      handlePlay={handlePlay}
      handlePause={handlePause}
      handleEnded={handleEnded}
      handleDuration={handleDuration}
    />
  </VideoBox>
);

Video.propTypes = {
  show: PropTypes.bool.isRequired,
  link: PropTypes.string.isRequired,
  playing: PropTypes.bool,
  loop: PropTypes.bool,
  muted: PropTypes.bool,
  controls: PropTypes.bool,
  pictureLink: PropTypes.string,
  pictureTitle: PropTypes.string,
  pictureAlt: PropTypes.string,
  handleProgress: PropTypes.func,
  handlePlay: PropTypes.func,
  handlePause: PropTypes.func,
  handleEnded: PropTypes.func,
  handleDuration: PropTypes.func,
};

export default Video;
