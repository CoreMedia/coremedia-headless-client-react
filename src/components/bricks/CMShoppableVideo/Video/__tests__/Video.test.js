// @flow
import React from 'react';

import Video from '..';

describe('Video Component', () => {
  it('should render correctly and proxy link prop', () => {
    const wrapper = shallow(<Video show link="coremedia:///media/6200/data" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render correctly and proxy pictureLink, pictureAlt and pictureTitle props', () => {
    const wrapper = shallow(
      <Video
        show
        link="coremedia:///media/6200/data"
        pictureLink="coremedia:///image/6188/data"
        pictureTitle="Title"
        pictureAlt="Alt"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render correctly and proxy playing prop', () => {
    const wrapper = shallow(<Video show link="coremedia:///media/6200/data" playing={false} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render correctly and proxy loop prop', () => {
    const wrapper = shallow(<Video show link="coremedia:///media/6200/data" loop={false} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render correctly and proxy muted prop', () => {
    const wrapper = shallow(<Video show link="coremedia:///media/6200/data" muted={false} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render correctly and proxy controls prop', () => {
    const wrapper = shallow(<Video show link="coremedia:///media/6200/data" controls={false} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should proxy handler props', () => {
    const handleProgress = jest.fn().mockName('handleProgress');
    const handlePlay = jest.fn().mockName('handlePlay');
    const handlePause = jest.fn().mockName('handlePause');
    const handleEnded = jest.fn().mockName('handleEnded');
    const handleDuration = jest.fn().mockName('handleDuration');
    const wrapper = shallow(
      <Video
        show
        link="coremedia:///media/6200/data"
        pictureLink="coremedia:///image/6188/data"
        pictureTitle="Title"
        pictureAlt="Alt"
        playing={false}
        loop={false}
        muted={false}
        controls={true}
        handleProgress={handleProgress}
        handlePlay={handlePlay}
        handlePause={handlePause}
        handleEnded={handleEnded}
        handleDuration={handleDuration}
      />
    );
    const videoBrick = wrapper.find('VideoBrick');
    expect(videoBrick.prop('handleProgress')).toEqual(handleProgress);
    expect(videoBrick.prop('handlePlay')).toEqual(handlePlay);
    expect(videoBrick.prop('handlePause')).toEqual(handlePause);
    expect(videoBrick.prop('handleEnded')).toEqual(handleEnded);
    expect(videoBrick.prop('handleDuration')).toEqual(handleDuration);
  });
});
