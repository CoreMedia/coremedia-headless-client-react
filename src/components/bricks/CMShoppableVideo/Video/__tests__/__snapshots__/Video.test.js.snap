// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Component should render correctly and proxy controls prop 1`] = `
<VideoBox
  show={true}
>
  <VideoBrick
    controls={false}
    link="coremedia:///media/6200/data"
    loop={false}
    muted={false}
    playing={false}
  />
</VideoBox>
`;

exports[`Video Component should render correctly and proxy link prop 1`] = `
<VideoBox
  show={true}
>
  <VideoBrick
    controls={true}
    link="coremedia:///media/6200/data"
    loop={false}
    muted={false}
    playing={false}
  />
</VideoBox>
`;

exports[`Video Component should render correctly and proxy loop prop 1`] = `
<VideoBox
  show={true}
>
  <VideoBrick
    controls={true}
    link="coremedia:///media/6200/data"
    loop={false}
    muted={false}
    playing={false}
  />
</VideoBox>
`;

exports[`Video Component should render correctly and proxy muted prop 1`] = `
<VideoBox
  show={true}
>
  <VideoBrick
    controls={true}
    link="coremedia:///media/6200/data"
    loop={false}
    muted={false}
    playing={false}
  />
</VideoBox>
`;

exports[`Video Component should render correctly and proxy pictureLink, pictureAlt and pictureTitle props 1`] = `
<VideoBox
  show={true}
>
  <VideoBrick
    controls={true}
    link="coremedia:///media/6200/data"
    loop={false}
    muted={false}
    pictureAlt="Alt"
    pictureLink="coremedia:///image/6188/data"
    pictureTitle="Title"
    playing={false}
  />
</VideoBox>
`;

exports[`Video Component should render correctly and proxy playing prop 1`] = `
<VideoBox
  show={true}
>
  <VideoBrick
    controls={true}
    link="coremedia:///media/6200/data"
    loop={false}
    muted={false}
    playing={false}
  />
</VideoBox>
`;
